### DISCUSSION TITLE
Auto-updating file names

### DISCUSSION THREAD

**Discussion by MilanNedoma** [bonsai-rx/bonsai#1611] [Q&A] [answered]
Hi everyone,

I have looked up a way to set up the FileNames to be changed whenever necessary by only updating a single node, as a BasePath.
However, I am unable to run this workflow now, as it does not allow me to leave the .avi format written in the VideoCapture node.

Is there a way for me to keep the function of updating the FileNames by changing data in a single node, but also make sure that the suffix .avi in case of VideoCapture or .csv in case of CsvWriter?

Thank you very much in advance and have a great rest of the day.

[Workflow.txt](https://github.com/bonsai-rx/bonsai/files/13267250/Workflow.txt)

**Comment by bruno-f-cruz** [comment, marked as answer]
You problem is mostly branching. As [stated here](https://bonsai-rx.org/docs/articles/subjects.html#branching-subjects), branching delays the subscription to the original sequence. As a result your properties are not being set right from the start. I re-wrote your workflow to not use branches, lmk if it works!
[Workflow.zip](https://github.com/bonsai-rx/bonsai/files/13267954/Workflow.zip)

**Reply by MilanNedoma** [reply]
Thank you very much!
It did not fix the whole problem, as it was still having issues with the wrong suffix, however I just added a python transform node to rewrite the .avi into .csv when using the CsvWriter node. However I am very thankful for the help and for demonstration of branching reduction.
